integrations/amazon_bedrock/src/haystack_integrations/components/downloaders/s3/s3_downloader.py

```diff
@@ -0,0 +1,239 @@
+
+    def warm_up(self) -> None:
+        """Warm up the component by initializing the settings and storage."""
+        if self.storage is None:
+            settings = S3DownloaderSettings.from_env()
+
+            self.file_root_path.mkdir(parents=True, exist_ok=True)
+
+            self.storage = S3Storage(
+                settings.s3_bucket,
+                settings.s3_folder,
+                endpoint_url=settings.aws_endpoint_url,
+                session=self.session,
+            )
+
+    @component.output_types(documents=List[Document])
+    def run(
+        self,
+        documents: Optional[List[Document]] = None,
+    ) -> Dict[str, List[Document]]:
+        """Download S3-linked files and return enriched `Document`s.
+
+        :param documents: Documents to download. 
+        :returns: A dictionary with:
+            - `documents`: The downloaded `Document`s; each has `meta['file_path']` and may include `mime_type`.
+        :raises AmazonS3Error: If a download attempt fails.
+        :raises ValueError: If an S3 URL in inputs is invalid.
+        """
+    
+        if not documents:
+            return {"documents": []}
+
+        filtered_documents = self._filter_documents_by_extensions(documents)
+
+        try:
+            max_workers = min(self.max_workers, len(filtered_documents))
+            with ThreadPoolExecutor(max_workers=max_workers) as executor:
+                iterable = executor.map(self.download_file, filtered_documents)
+        finally:
+            self.cleanup_cache(filtered_documents)
+
+        downloaded_documents = [d for d in iterable if d is not None]
+        return {"documents": downloaded_documents}
+
+    def _filter_documents_by_extensions(self, documents: List[Document]) -> List[Document]:
+            return [doc for doc in documents if Path(doc.meta.get("file_name", "")).suffix.lower() in self.file_extensions]
+
+    def download_file(self, document: Document) -> Optional[Document]:
+        """
+        Download a single S3 object into the local file system and enrich the `Document`.
+
+        :param document: `Document` with file_id in the meta field.
+        :returns:
+            The same `Document` with `meta` containing the path of the
+            downloaded file and the mime type or `None` if skipped.
+        :raises AmazonS3Error: If the download or head request fails.
+        """
+        if self.storage is None:
+            raise RuntimeError(
+                f"The component {self.__class__.__name__} was not warmed up. "
+                """Call "warm_up()" before calling "download_file()"."""
+            )
+
+        try:
+            file_id = UUID(document.meta["file_id"])
+        except KeyError:
+            logger.warning(f"Document with ID {document.id!r} does not have a file_id in the meta field", exc_info=True)
+            return None
+
+        file_name = Path(document.meta.get("file_name", ""))
```

Comment:
just a nit. But maybe worth the little tweak to make this more generic: how about making this meta key configurable?